homeassistant/helpers/storage.py

```diff
@@ -66,7 +67,7 @@ def __init__(
         key: str,
         private: bool = False,
         *,
-        encoder: Optional[Type[JSONEncoder]] = None,
+        encoder: Optional[Type[JSONEncoder]] = HelperJSONEncoder,
```

Comment:
We should not update the default for ALL stores in Home Assistant to avoid passing in an encoder for one instance. Revert this.